backend/account_v2/authentication_service.py

```diff
@@ -295,36 +303,108 @@ def get_organization_by_org_id(self, id: str) -> OrganizationData:
+            self._update_user_credentials(user)
+
+            return True
+
+        except Exception as e:
+            logger.error(f"Failed to set default user: {str(e)}")
+            return False
+
+    def _get_or_create_user(self, organization: Organization) -> User:
+        """Get existing user or create a new one based on organization context.
 
         Args:
-            username (str): The username of the default user.
-            password (str): The password of the default user.
+            organization: The organization context
 
         Returns:
-            bool: True if the default user is successfully created and saved,
-                False otherwise.
+            User: The retrieved or created user
         """
-        if (
-            username != DefaultOrg.MOCK_USER
-            or password != DefaultOrg.MOCK_USER_PASSWORD
-        ):
-            return False
+        if not organization:
+            return self._create_mock_user()
 
+        return self._get_or_create_organization_user()
+
+    def _create_mock_user(self) -> User:
+        """Create a new mock user if it doesn't exist.
+
+        Returns:
+            User: The created or existing mock user
+        """
         user, created = User.objects.get_or_create(username=DefaultOrg.MOCK_USER)
         if created:
-            user.password = make_password(DefaultOrg.MOCK_USER_PASSWORD)
-        else:
-            user.user_id = DefaultOrg.MOCK_USER_ID
-            user.email = DefaultOrg.MOCK_USER_EMAIL
-            user.password = make_password(DefaultOrg.MOCK_USER_PASSWORD)
+            logger.info(f"Created new user with username {DefaultOrg.MOCK_USER}")
+        return user
+
+    def _get_or_create_organization_user(self) -> User:
+        """Get or create an organization user with admin privileges.
+
+        Returns:
+            User: The organization user with admin role
+        """
+        admin_user = self._get_admin_user()
+        if admin_user:
+            return admin_user
+
+        member = self._promote_first_member_to_admin()
+        if member:
+            return member.user
+
+        return self._create_mock_user()
+
+    def _get_admin_user(self) -> Optional[User]:
+        """Get the first admin user from the organization.
+
+        Returns:
+            Optional[User]: The admin user if exists, None otherwise
+        """
+        admin_members = OrganizationMemberService.get_members_by_role(
+            role=UserRole.ADMIN.value
+        )
+        return admin_members[0].user if admin_members else None
+
+    def _promote_first_member_to_admin(self) -> Optional[OrganizationMember]:
+        """Promote the first organization member to admin role.
+
+        Returns:
+            Optional[OrganizationMember]: The promoted member if exists, None otherwise
+        """
+        members = OrganizationMemberService.get_members()
+        if not members:
+            logger.error("No organization member found")
```

Comment:
NIT: Will this error log be thrown for a fresh installation? If that's the case we should either mark it a warning or indicate in the log that its not an error somehow

Reply:
updated as a warning